src/controllers/FearConditioningModuleViewController.ts

```diff
@@ -116,20 +123,56 @@ export class FearConditioningModuleViewController extends GenericModuleViewContr
     return trials
   }
 
+  /**
+   * Called when the user skips two consecutive trials
+   *
+   * @param experimentVC
+   */
+  onSkip(experimentVC: ExperimentViewController) {
+    Alert.alert(
+      'Attention Required',
+      'You have not been rating trials in the designated time, Please try to answer them as fast as you can.',
+      [
+        {
+          text: 'Exit Experiment',
+          onPress: () => experimentVC.screenOutParticipant(),
+          style: 'cancel',
+        },
+        {
+          text: 'Continue to next trial',
+          onPress: () => this.render(experimentVC),
+        },
+      ],
+      { cancelable: false },
+    )
+  }
+
   /**
    * Called when the user submits their response.
+   *
+   * @param response
+   * @param experimentVC
    */
-  onSubmit(value: number, experimentVC: ExperimentViewController) {
+  onSubmit(
+    response: FearConditioningTrialResponse,
+    experimentVC: ExperimentViewController,
+  ) {
     // Set the users rating
-    this.moduleState.trials[this.currentTrialIndex].rating = value
+    this.moduleState.trials[this.currentTrialIndex].response = response
     // Move to next trial
     this.currentTrialIndex = this.currentTrialIndex + 1
     // Save changes
     this.saveState()
     // If we have run out of trials then call next module
     if (this.currentTrialIndex === this.moduleState.trials.length) {
-      experimentVC.onModuleComplete()
+      return experimentVC.onModuleComplete()
```

Comment:
Is it correct to return a value here? (Is the `onSubmit` method supposed to return a value?)

Reply:
Ah meant to get rid of this, switched lower `if` to an `else if`